### PR TITLE
drop `--hostname` from CLI

### PR DESCRIPTION
**Pull request by rgildein**
After discussion, we agreed that this option is not needed, since it can be confusing which hostname the user meant.

### CODE REVIEW

File: tests/unit/test_commands.py [hunk 1]
```diff
@@ -233,57 +228,10 @@ def test_parse_args_quiet_verbose_exclusive(args):
...
-                force=True,
-                machines=None,
-                hostnames={"1", "2", "3"},
-                availability_zones=None,
-                **{"upgrade_group": "data-plane"}
-            ),
-        ),
-        (
-            # repetitive machine 3
-            ["plan", "data-plane", "--hostname=1,2,3", "--force", "-n=3,4"],
```
Comment (gabrielcocenza) [inline tests/unit/test_commands.py:273]: I think we can keep this test and just change `hostname`to `machine`. This test is also checking that no repetitive elements appears after the parding

File: tests/unit/test_commands.py [hunk 2]
```diff
@@ -554,115 +494,10 @@ def test_parse_args_plan(args, expected_CLIargs):
...
-                backup=True,
-                force=True,
-                machines=None,
-                hostnames={"1", "2", "3"},
-                availability_zones=None,
-                **{"upgrade_group": "data-plane"}
-            ),
-        ),
-        (
-            # repetitive machine 3
```
Comment (gabrielcocenza) [inline tests/unit/test_commands.py:643]: Same here

**Review by gabrielcocenza — COMMENTED** [review]
Just two small comments, but LGTM

**Comment by gabrielcocenza** [comment]
/canonical/self-hosted-runners/run-workflows 56f977c0aa975c3b9ff32f2b5d43281da3e6420d

**Review by agileshaw — COMMENTED** [review]
+1 on Gabriel's suggestion of adding the repetitive tests back. Otherwise LGTM

**Review by gabrielcocenza — APPROVED** [review]

**Review by agileshaw — APPROVED** [review]
LGTM

**Pull request merged**